scripts/getSnippet.js

```diff
@@ -0,0 +1,190 @@
+    } else {
+      console.log('Matched')
+    }
+  }
+
+  return { snippets, collectionHash }
+}
+
+/**
+ * @description adds or updates the hash beside the snippet marker
+ *
+ * @input filePath
+ *
+ * @returns void
+ */
+function addHashToFile(filePath) {
+  const fileContent = fs.readFileSync(filePath, 'utf-8')
+  const regex =
+    /(<!--snippet)(?:\s+([a-f0-9]{32}))?(-->)([\s\S]*?)(<!--\/snippet-->)/g
+
+  let newContent = fileContent.replace(
+    regex,
+    (match, start, existingHash, end, content, closeTag) => {
+      // Check if the content follows the expected format
+      const formatRegex =
+        /^\s*(<!--title:\s*.*?-->\s*)?\s*(<!--descr:\s*.*?-->\s*)?\s*```(\w+)\s*\n([\s\S]*?)\n\s*```\s*$/
+      const formatMatch = formatRegex.exec(content)
+
+      if (formatMatch) {
+        const [, title, desc, language, snippetContent] = formatMatch
+        const snippetHash = hashSnippet(snippetContent.trim())
+        return `${start} ${snippetHash}${end}${content}${closeTag}`
+      } else {
+        // If the format doesn't match, return the original content without modification
+        return match
+      }
+    },
+  )
+
+  fs.writeFileSync(filePath, newContent, 'utf-8')
+}
+
+/**
+ * @description cleans the title and description fields by removing the tags
+ *
+ * @input snippets
+ *
+ * @returns List of cleaned snippets
+ */
+function cleanSnippets(snippets) {
+  return snippets.map((snippet) => ({
+    ...snippet,
+    title: snippet.title.replace(/<!--title:\s*(.*?)-->/, '$1').trim(),
+    desc: snippet.desc.replace(/<!--descr:\s*(.*?)-->/, '$1').trim(),
+  }))
+}
+
+/**
+ * @description Main function that extracts snippets, cleans them, and adds hashes to the file
+ *
+ * @input filePath
+ *
+ * @returns void
+ */
+async function processSnippets(filePath) {
+  try {
+    const { snippets, collectionHash } = extractSnippetsFromFile(filePath)
+    const cleanedSnippets = cleanSnippets(snippets)
+    console.log(cleanedSnippets)
+    addHashToFile(filePath)
+    console.log('Hashes added to file')
+
+    // prepare the payload
+    const payload = {
+      hashes: collectionHash,
+      snippets: cleanedSnippets,
+    }
+    console.log('hash', payload.hashes)
+    console.log('code', payload.snippets)
+
+    // send to the server
+    const response = await fetch(
+      'https://kb-backend-ompt.onrender.com/api/code_snippet',
+      {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(payload),
+      },
+    )
+
+    if (!response.ok) {
+      throw new Error('Failed to send snippets to the server')
+    }
+    console.log('Snippets sent successfully')
+  } catch (error) {
+    console.error(error.message)
+  }
+}
+
+/**
+ * @description Function to get all markdown file paths in a directory
+ *
+ * @input dir
+ * @input fileList
+ *
+ * @returns List of markdown file paths
+ */
+function getMdPaths(dir, fileList = []) {
```

Comment:
An existing `getMdPaths`  function exists in the `scripts` folder. Can we import that instead of redeclaring here?